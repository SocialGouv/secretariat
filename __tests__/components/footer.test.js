import { render } from "@testing-library/react"

import Footer from "../../src/components/footer"

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

process.env.NEXT_PUBLIC_APP_VERSION = "0.0.42"
process.env.NEXT_PUBLIC_APP_VERSION_COMMIT =
  "2f7c23024eeb820ed67aef00bc40e7955111cbf4"
process.env.NEXT_PUBLIC_APP_REPOSITORY_URL =
  "https://github.com/SocialGouv/secretariat"

it("renders footer", () => {
  const { container } = render(<Footer />)
  expect(container).toMatchSnapshot()
})
