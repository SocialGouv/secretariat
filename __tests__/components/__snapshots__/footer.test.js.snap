// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders footer 1`] = `
<div>
  <footer>
    <div
      class="footer-content"
    >
      <div
        class="logo-info"
      >
        <div
          class="logo big"
        >
          République
          <br />
          Française
        </div>
        <div
          class="info"
        >
          <p>
            <strong>
              Secretariat
            </strong>
             est une application permettant de gérer les accès des utilisateurs aux services de la Fabrique Numérique.
          </p>
          <p>
            Pour en savoir plus sur l'utilisation de
             
            <strong>
              Secretariat
            </strong>
            , visitez le
             
            <a
              href="https://github.com/SocialGouv/www/wiki/Inscrire-son-%C3%A9quipe-au-standup"
              rel="noopener noreferrer"
              target="_blank"
            >
              wiki
            </a>
            .
          </p>
          <p
            class="flex items-center"
          >
            <i
              class="ri-github-fill ri-xl relative top-px mr-1"
            />
            version 
            0.0.42
             (
            <a
              href="https://github.com/SocialGouv/secretariat/tree/2f7c23024eeb820ed67aef00bc40e7955111cbf4"
              rel="noreferrer"
              target="_blank"
            >
              2f7c2302
            </a>
            )
          </p>
        </div>
      </div>
      <div
        class="links"
      >
        <a
          href="https://www.fabrique.social.gouv.fr/"
          rel="noreferrer"
          target="_blank"
        >
          La Fabrique Numérique
        </a>
        <a
          href="/legal"
        >
          Mentions légales
        </a>
        <button
          aria-label="theme toggle"
        >
          <i
            class="theme-toggle-button ri-sun-line"
          />
        </button>
      </div>
    </div>
  </footer>
</div>
`;
